devopstools-jenkins-chefhabitat-terraform/azuredeploy.json

```diff
@@ -0,0 +1,391 @@
+                    },
+                    "AzureTenantId": {
+                        "value": "[variables('azureTenantId')]"
+                    },
+                    "deploymentApiVersion": {
+                        "value": "[variables('deploymentApiVersion')]"
+                    },
+                    "baseUrl": {
+                        "value": "[variables('baseUrl')]"
+                    }
+                }
+            }
+        },
+        {
+            "comments": "This is the deployment of chef habitat related nested resources",
+            "type": "Microsoft.Resources/deployments",
+            "name": "BuildInstance",
+            "apiVersion": "2016-02-01",
+            "dependsOn": [
+                "[concat('Microsoft.Network/virtualNetworks/',variables('networkSettings').virtualNetworkName)]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('baseUrl'),'/nested/build-instance.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "buildInstanceSettings": {
+                        "value": "[variables('buildInstanceSettings')]"
+                    },
+                    "apiVersions": {
+                        "value": "[variables('apiVersions')]"
+                    },
+                    "networkSettings": {
+                        "value": "[variables('networkSettings')]"
+                    },
+                    "AzureSubscriptionId": {
+                        "value": "[variables('AzureSubscriptionId')]"
+                    },
+                    "AzureApplicationId": {
+                        "value": "[parameters('AzureApplicationId')]"
+                    },
+                    "AzureClientSecret": {
+                        "value": "[parameters('AzureClientSecret')]"
+                    },
+                    "AzureTenantId": {
+                        "value": "[variables('AzureTenantId')]"
+                    },
+                    "packerStorageAccName": {
+                        "value": "[variables('packerStorageAccountName')]"
+                    }
+                }
+            }
+        }
+    ],
+    "outputs": {
+        "ResourceGroupName": {
+            "type": "string",
+            "value": "[variables('resourceGroupName')]"
+        },
+        "VirtualNetworkName": {
+            "type": "string",
+            "value": "[variables('networkSettings').virtualNetworkName]"
+        },
+        "JenkinsFQDN": {
+            "type": "string",
+            "value": "[reference('Jenkins').outputs.jenkinsDNS.value]"
+        },
+        "JenkinsWebUIURL": {
+            "type": "string",
+            "value": "[concat(reference('Jenkins').outputs.jenkinsDNS.value,':8080')]"
+        },
+        "kibanaWebUIUsername": {
```

Comment:
why are we hardcoding outputs... particularly secrets?

Reply:
Fixed